pandas/core/generic.py

```diff
@@ -6041,9 +6048,23 @@ def sample(
         if weights is not None:
             weights = sample.preprocess_weights(self, weights, axis)
 
-        sampled_indices = sample.sample(obj_len, size, replace, weights, rs)
+        sampled_indices = sample.sample(
+            obj_len=obj_len,
+            size=size,
+            replace=replace,
+            n_samples=n_samples,
+            weights=weights,
+            random_state=rs,
+        )
         result = self.take(sampled_indices, axis=axis)
 
+        if n_samples > 1:
+            sample_no = np.repeat(np.arange(n_samples), size)
+
+            sampled_indices = np.reshape(sampled_indices, (n_samples, size))
+
+            result = result.assign(sample_no=sample_no)
```

Comment:
I think `replicate` or `sample` could be more suitable names for this new column, what do you think?